Theme: Fix placeholder image references

EricDunsworth opened the pull request:
Some of GCWeb's code samples from recent years were referencing placeholder images from [placeholder.com](https://placeholder.com/) and [placehold.it](https://placehold.it/). But those sites underwent changes in ownership in March 2023 and consequently ceased serving placeholder images.

This deals with it by replacing all references to via.placeholder.com and placehold.it with dummyimage.com (and swaps ? characters with & where custom text is specified to fit the latter's URL writing scheme).

GCWeb has historically used [dummyimage.com](https://dummyimage.com/) and that site still exists to this day.

Left the following references to [placeholder.com](https://placeholder.com/) "as-is" in the following WCAG assessments since they're meant to represent specific points in time:
* [templates/institutional-landing/reports/a11y-1.json](https://github.com/wet-boew/GCWeb/blob/master/templates/institutional-landing/reports/a11y-1.json)
* [templates/institutional-landing/reports/a11y-2.json](https://github.com/wet-boew/GCWeb/blob/master/templates/institutional-landing/reports/a11y-2.json)

EricDunsworth commented on design-patterns/gc-intro/index.json-ld:
Just so these are noted...
* This file's changes don't currently add an extra iteration for the revised example... might try force-pushing an update to add it at some point
* I'm unsure if docs examples are supposed to contain full image URLs (could've sworn I saw some others out there that used something along the lines of ``src="[your-image]"`` instead of a placeholder URL)

Garneauma replied:
The dummy images URL aren't part of the API, so no need to create an iteration.

Garneauma commented on design-patterns/gc-intro/index.json-ld:
```diff
@@ -155,6 +155,10 @@
 				"_:implement_intro_half_width_image"
 			],
 			"history": [
+				{
+					"en": "November 2024 - Fixed references to placeholder image (only affected documentation's code samples).",
+					"fr": "Novembre 2024 - Correction de références vers les espaces réservés pour des images (affectait seulement les exemples de code dans la documentation)."
+				},
```
Please remove this history item.

EricDunsworth replied:
Done :).

EricDunsworth committed:
Theme: Fix placeholder image references

Some of GCWeb's code samples from recent years were referencing placeholder images from placeholder.com and placehold.it. But those sites underwent changes in ownership in March 2023 and consequently ceased serving placeholder images.

This deals with it by replacing all references to via.placeholder.com and placehold.it with dummyimage.com (and swaps ? characters with & where custom text is specified to fit the latter's URL writing scheme).

GCWeb has historically used dummyimage.com and that site still exists to this day.

Left the following references to placeholder.com "as-is" in the following WCAG assessments since they're meant to represent specific points in time:
* templates/institutional-landing/reports/a11y-1.json
* templates/institutional-landing/reports/a11y-2.json
